core/api/src/app/wallets/get-pending-transactions-by-addresses.ts

```diff
@@ -0,0 +1,25 @@
+import { CouldNotFindError } from "@/domain/errors"
+
+import { WalletOnChainPendingReceiveRepository } from "@/services/mongoose"
+
+export const getPendingTransactionsForWalletsByAddresses = async ({
```

Comment:
I know wallets are passed from source but we must enforce in app layer that wallets belong the user/account (similar to logic in `getPendingOnChainTransactionsForAccountByWalletIds`)

Reply:
I dont necessarily agree with that. The function does what it says which is `getPendingTransactionsForWalletsByAddress`. It's different than the `getPendingOnChainTransactionsForAccountByWalletIds` which says its getting them for a specific account and therefore must verify that the account owns the wallets. I think this is a pretty common pattern across the app layer.